Jenkinsfile

```diff
@@ -0,0 +1,630 @@
+                'dse-6.7', // Current DataStax Enterprise
+                'dse-6.8', // Development DataStax Enterprise
+                'ALL'],
+      description: '''Apache Cassandra and DataStax Enterprise server version to use for adhoc <b>BUILD-AND-EXECUTE-TESTS</b> <strong>ONLY!</strong>
+                      <table style="width:100%">
+                        <col width="15%">
+                        <col width="85%">
+                        <tr>
+                          <th align="left">Choice</th>
+                          <th align="left">Description</th>
+                        </tr>
+                        <tr>
+                          <td><strong>2.1</strong></td>
+                          <td>Apache Cassandra v2.1.x</td>
+                        </tr>
+                        <tr>
+                          <td><strong>3.11</strong></td>
+                          <td>Apache Cassandra v3.11.x</td>
+                        </tr>
+                        <tr>
+                          <td><strong>4.0</strong></td>
+                          <td>Apache Cassandra v4.x (<b>CURRENTLY UNDER DEVELOPMENT</b>)</td>
+                        </tr>
+                        <tr>
+                          <td><strong>dse-5.1</strong></td>
+                          <td>DataStax Enterprise v5.1.x</td>
+                        </tr>
+                        <tr>
+                          <td><strong>dse-6.0</strong></td>
+                          <td>DataStax Enterprise v6.0.x</td>
+                        </tr>
+                        <tr>
+                          <td><strong>dse-6.7</strong></td>
+                          <td>DataStax Enterprise v6.7.x</td>
+                        </tr>
+                        <tr>
+                          <td><strong>dse-6.8</strong></td>
+                          <td>DataStax Enterprise v6.8.x (<b>CURRENTLY UNDER DEVELOPMENT</b>)</td>
+                        </tr>
+                      </table>''')
+    booleanParam(
+      name: 'ADHOC_BUILD_AND_EXECUTE_TESTS_EXECUTE_EXAMPLES',
+      defaultValue: false,
+      description: 'Flag to determine if the examples should be executed for adhoc builds')
+    booleanParam(
+      name: 'JUNIT_REPORT_STACK',
+      defaultValue: true,
+      description: 'Flag to determine if stack trace should be enabled with test failures for scheduled or adhoc builds')
+    booleanParam(
+      name: 'TEST_TRACE',
+      defaultValue: true,
+      description: 'Flag to determine if test tracing should be enabled for scheduled or adhoc builds')
+    choice(
+      name: 'CI_SCHEDULE',
+      choices: ['DO-NOT-CHANGE-THIS-SELECTION', 'WEEKNIGHTS'],
+      description: 'CI testing schedule to execute periodically scheduled builds and tests of the driver (<strong>DO NOT CHANGE THIS SELECTION</strong>)')
+  }
+
+  triggers {
+    parameterizedCron("""
+      # Every weeknight (Monday - Friday) around 3:00 AM
+      H 3 * * 1-5 %CI_SCHEDULE=WEEKNIGHTS
+    """)
```

Comment:
if you want to have nightlies on `master` only you need to add a check here (e.g. https://github.com/datastax/csharp-driver/blob/master/Jenkinsfile#L429)

note that nightly runs will occur even if no changes are pushed so if there is no check here, it will run a nightly build for every branch that the repo has if I'm not mistaken

Reply:
Great catch! I only want to build `master` for weeknights .